### ISSUE TITLE
Document headache approach in README

### ISSUE THREAD

**Issue by fbiville** [fbiville/headache#26]
This is mainly explained [here](https://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated).

TL;DR: 

 - copyright year range start: it should be based on the first introduction of the source file (could predate the repository itself)
 - copyright year range end (or single year): it should be based on the last **meaningful** change

Obviously, `headache` cannot distinguish meaningful changes from other kinds of change. However, `headache` bases this information on the last VCS change (at the moment, only `Git` is supported) and it is up to the user to determine whether that change is indeed meaningful (then accepts the `headache` change and commits it) or not (then alters/discards the change).
